core/src/main/scala/com/mesosphere/usi/core/matching/PortResource.scala

```diff
@@ -0,0 +1,223 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.ResourceUtil
+import com.mesosphere.usi.core.matching.PortResource._
+import com.mesosphere.usi.core.models.{ResourceMatchResult, ResourceRequirement, ResourceType}
+import com.mesosphere.usi.core.protos.ProtoBuilders
+import org.apache.mesos.v1.{Protos}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+case class PortResource(ports: Seq[Int], random: Random = Random) extends ResourceRequirement {
+  override def description: String = s"$resourceType:[${ports.mkString(",")}]"
+
+  override def resourceType: ResourceType = ResourceType.PORTS
+
+  @tailrec private def findResourceToMatch(
+      unmatchedResources: List[Protos.Resource],
+      remainingResources: List[Protos.Resource]): Option[ResourceMatchResult] = {
+
+    remainingResources match {
+      case Nil =>
+        None
+      case next :: rest =>
+        matchPortResource(next) match {
+          case Nil =>
+            findResourceToMatch(next :: unmatchedResources, rest)
+          case consumedResources =>
+            Some(
+              ResourceMatchResult(
+                consumedResources,
+                unmatchedResources ++ rest ++ ResourceUtil.consumeResources(Seq(next), consumedResources)))
```

Comment:
Where is `consumerResources` defined?

Reply:
in ResourceUtil. That class was already ported over with Tests by Tim in previous PR